napari/_qt/qt_color_dialog.py

```diff
@@ -0,0 +1,267 @@
+        self.setStyleSheet('#colorSwatch {background-color: ' + color + ';}')
+
+    def mouseReleaseEvent(self, event):
+        """Show QColorPopup picker when the user clicks on the swatch."""
+        if event.button() == Qt.LeftButton:
+            initial = QColor(*(255 * self._color).astype('int'))
+            popup = QColorPopup(self, initial)
+            popup.colorSelected.connect(self.setColor)
+            popup.show_right_of_mouse()
+
+    def setColor(self, color: ColorType) -> None:
+        """Set the color of the swatch.
+
+        Parameters
+        ----------
+        color : ColorType
+            Can be any ColorType recognized by our
+            utils.colormaps.standardize_color.transform_color function.
+        """
+        try:
+            _color = transform_color(color)[0]
+        except ValueError:
+            _color = UNKNOWN
+        emit = np.any(self._color != _color)
+        self._color = _color
+        if emit or np.all(_color == TRANSPARENT):
+            self.color_changed.emit(_color)
+
+
+class QColorLineEdit(QLineEdit):
+    """A LineEdit that takes hex, rgb, or autocompletes common color names."""
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self._compl = QCompleter(list(get_color_dict()))
+        self._compl.setCompletionMode(QCompleter.InlineCompletion)
+        self.setCompleter(self._compl)
+        self.setTextMargins(2, 2, 2, 2)
+
+    def setText(self, color: ColorType):
+        """Set the text of the lineEdit using any ColorType.
+
+        Colors will be converted to standard SVG spec names if possible,
+        or shown as #RGBA hex if not.
+
+        Parameters
+        ----------
+        color : ColorType
+            Can be any ColorType recognized by our
+            utils.colormaps.standardize_color.transform_color function.
+        """
+        if isinstance(color, int) and color == UNKNOWN:
+            text = "⚠️ unknown"
+        else:
+            _rgb = transform_color(color)[0]
+            _hex = rgb_to_hex(_rgb)[0]
+            text = hex_to_name.get(_hex, _hex)
+        super().setText(text)
+
+
+class CustomColorDialog(QColorDialog):
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self.setObjectName('CustomColorDialog')
+
+    def keyPressEvent(self, event):
+        event.ignore()
+
+
+class QColorPopup(QtPopup):
+    """A QColorDialog inside of our QtPopup.
+
+    Allows all of the show methods of QtPopup (like show relative to mouse).
+    Passes through signals from the ColorDialogm, and handles some keypress
+    events.
+    """
+
+    currentColorChanged = Signal(QColor)
+    colorSelected = Signal(QColor)
+
+    def __init__(self, parent=None, initial=None):
+        super().__init__(parent)
+        self.setObjectName('QtColorPopup')
+        self.color_dialog = CustomColorDialog(self)
+
+        # TODO: ask everyone how they feel about native vs QtColorDialog look
```

Comment:
I like QtColorDialog, gives us more flexibility in the long run and is how illustrator looks. As noted in the main thread, we could get help from a designer in the future too